agent/agent.go

```diff
@@ -1485,15 +1485,25 @@ func writeFileAtomic(path string, contents []byte) error {
 		return err
 	}
 	if _, err := fh.Write(contents); err != nil {
+		fh.Close()
+		os.Remove(tempPath)
 		return err
 	}
 	if err := fh.Sync(); err != nil {
+		fh.Close()
+		os.Remove(tempPath)
 		return err
 	}
 	if err := fh.Close(); err != nil {
+		fh.Close()
```

Comment:
Apparently not: https://stackoverflow.com/questions/15348431/does-close-call-fsync-on-linux